File: bungee/src/main/java/com/emanuelvini/nuven/core/bungee/registry/ConfigurationRegistry.java
package com.emanuelvini.nuven.core.bungee.registry;

import com.emanuelvini.nuven.core.bungee.BungeeMain;
import com.emanuelvini.nuven.core.bungee.configuration.database.DatabaseValue;
import com.henryfabio.minecraft.configinjector.bungee.injector.BungeeConfigurationInjector;
import lombok.AllArgsConstructor;
import lombok.val;

@AllArgsConstructor
public class ConfigurationRegistry {

    private BungeeMain plugin;

    public void register() {
        val bungeeInjector = new BungeeConfigurationInjector(plugin);

        bungeeInjector.saveDefaultConfiguration(plugin,
                "database.yml"
                );

        bungeeInjector.injectConfiguration(
                DatabaseValue.instance()
        );
    }

}

File: bungee/src/main/kotlin/com/emanuelvini/nuven/core/bungee/registry/primary/ConfigurationRegistry.kt
package com.emanuelvini.nuven.core.bungee.registry.primary

import com.emanuelvini.nuven.core.bungee.BungeeMain
import com.emanuelvini.nuven.core.bungee.configuration.ChatValue
import com.emanuelvini.nuven.core.bungee.configuration.LanguageValue
import com.emanuelvini.nuven.core.shared.database.DatabaseValue
import com.henryfabio.minecraft.configinjector.bungee.injector.BungeeConfigurationInjector


class ConfigurationRegistry (
    private val plugin: BungeeMain
) {

    fun register() {
        val bungeeInjector = BungeeConfigurationInjector(plugin)
        plugin.settingsFolder.mkdirs()
        bungeeInjector.saveDefaultConfiguration(
            plugin,
            "database.yml",
            "settings/language.yml",
            "settings/chat.yml"
        )
        bungeeInjector.injectConfiguration(
            DatabaseValue.instance,
            LanguageValue.instance,
            ChatValue.instance
        )
    }
}
